### PR TITLE
Add file.getEntryLink, isSkylinkV1, and isSkylinkV2

### PR DESCRIPTION
**Pull request by m-cat**

### CODE REVIEW

**Review by dghelm — APPROVED** [review]
Good, so long as it has been tested on a V2 Skylink and confirmed that the return is its dataLink.

File: src/skylink/sia.ts [hunk 1]
```diff
@@ -1,10 +1,29 @@
...
+
 import { hashAll } from "../crypto";
-import { encodePrefixedBytes } from "../utils/encoding";
+import { base64RawUrlToUint8Array, uint8ArrayToBase64RawUrl, encodePrefixedBytes } from "../utils/encoding";
 import { hexToUint8Array, isASCIIString, stringToUint8ArrayUtf8, trimSuffix } from "../utils/string";
 import { validateHexString, validateNumber, validateUint8ArrayLen } from "../utils/validation";
 
-// The raw size of the data that gets put into a link.
+/**
+ * The size of the Skylink after it has been encoded using base32.
```
Comment (dghelm) [inline src/skylink/sia.ts:9]: "string length" might be more clear than "size" here.

File: src/skylink/sia.ts [hunk 2]
```diff
@@ -1,10 +1,29 @@
...
+ */
+const BASE32_ENCODED_SKYLINK_SIZE = 55;
+
+/**
+ * The size of the Skylink after it has been encoded using base64.
```
Comment (dghelm) [inline src/skylink/sia.ts:14]: "string length"?

**Review by dghelm — APPROVED** [review]

File: src/skylink/sia.ts (hunk 1 above)
Comment (m-cat) [reply src/skylink/sia.ts]: These comments were copied from skyd. Fixed.

**Review by peterjan — APPROVED** [review]

**Review by ro-tex — APPROVED** [review]

**Pull request merged**